app/models/dataset.rb

```diff
@@ -21,6 +21,18 @@ class Dataset
                 :_index, :_type, :_id, :_score, :_source,
                 :_version
 
+  def datafiles
+    @datafiles.map { |file| Datafile.new(file) }
```

Comment:
Can't you lose the @ as there is an attr_accessor for datafiles?

Reply:
Also I think this means on the `views/datasets/show` page we need to have:

 
```
<% unless @dataset.datafiles%>
    <%= render partial: "timeseries_data", locals: { datafiles: @timeseries_datafiles } %>
  <% else %>
    <%= render partial: "non_timeseries_data", locals: { datafiles: @non_timeseries_datafiles } %>
  <% end %>
```


Because this current branch has this bug:


<img width="659" alt="screen shot 2017-09-14 at 13 30 23" src="https://user-images.githubusercontent.com/17908089/30429858-f5a748ba-9950-11e7-91c8-7a76d36907b6.png">

Reply:
`attr_accessor` gives me this getter method for free:

```
def datafiles
  @datafiles
end
```

I am overwriting this getter method so it returns an array of objects.